test/e2e/singlecluster/e2e_test.go

```diff
@@ -414,17 +402,13 @@ var _ = ginkgo.Describe("Kueue", func() {
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
-			expectJobUnsuspendedWithNodeSelectors(jobKey, map[string]string{
-				"instance-type": "on-demand",
+			ginkgo.By("ensuring that the job has been unsuspended", func() {
+				expectJobUnsuspendedWithNodeSelectors(jobKey, map[string]string{
+					"instance-type": "on-demand",
+				})
 			})
-			gomega.Eventually(func() bool {
-				if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
-					return false
-				}
-				return workload.HasQuotaReservation(createdWorkload) &&
-					apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished)
 
-			}, util.LongTimeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Expect(util.DeleteAllPodsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
 		})
 
 		ginkgo.It("Should suspend a job when its checks become invalid", func() {
```

Comment:
Missed that one! I can work on it and check if there are others. Thanks!

Reply:
Since this test doesn't wait for the job to finish, IMO there's no need to use here a binary that ends. That's why I've replaced it by the default `SampleJob` which uses the `pause` image